### PR TITLE
use xmap instead of vmap in textobjects

### PR DESCRIPTION
**Pull request by kuator**

### CODE REVIEW

**Comment by kuator** [comment]
`vmap` works for visual and selecte mode. `xmap`  works only for visual mode. Snippet plugins use select mode for editing placeholders, and it's extremely annoying when instead of inserting a character, a text object gets selected.

**Comment by vigoux** [comment]
Hey you, did you looked a bit elsewhere to see if we have other vmaps?

**Comment by kuator** [comment]
> Hey you, did you looked a bit elsewhere to see if we have other vmaps?

I have found one more place, my lua-fu is weak, so I just grepped for places where mappings are set.

**Review by vigoux — APPROVED** [review]
LGTM, @theHamsta this is textobjects related so...

File: lua/nvim-treesitter/incremental_selection.lua [hunk 1]
```diff
@@ -100,7 +100,7 @@ function M.attach(bufnr)
     if funcname == "init_selection" then
       mode = 'n'
     else
-      mode = 'v'
+      mode = 'x'
```
Comment (theHamsta) [inline lua/nvim-treesitter/incremental_selection.lua:103]: You missed changing the corresponding detach. But everything else LGTM! :+1:  Thanks!

Comment (kuator) [reply lua/nvim-treesitter/incremental_selection.lua:103]: sorry, fixed it.

**Review by vigoux — APPROVED** [review]

**Pull request merged**